spice_agent/hardware/actions.py

```diff
@@ -44,6 +47,137 @@ def get_darwin_system_profiler_values(self) -> Dict[str, str]:
             "apple_serial_number": data_type.get("serial_number"),
         }
 
+    def _is_nvidia_smi_available(self) -> bool:
+        """
+        Checks if nvidia-smi is avaialble
+        """
+        try:
+            nvidia_smi_command = "nvidia-smi"
+            subprocess.check_output(nvidia_smi_command)
+            return True
+        except (FileNotFoundError, subprocess.CalledProcessError):
+            return False
+
+    def _get_supported_metal_device(self) -> int:
+        """
+        Checks if metal hardware is supported. If so, the index
+        of the supported metal device is returned
+        """
+        supported_metal_device = -1
```

Comment:
not sure if we should be returning -1 if we dont find the value. "None" may be a better use
```suggestion
        supported_metal_device = None
```